ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/review/v1/review/clientlibs/site/js/reviewview.js

```diff
@@ -0,0 +1,322 @@
+/*******************************************************************************
+ * Copyright 2024 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+(function () {
+
+  "use strict";
+  class Review extends FormView.FormFieldBase {
+
+    static NS = FormView.Constants.NS;
+    /**
+     * Each FormField has a data attribute class that is prefixed along with the global namespace to
+     * distinguish between them. If a component wants to put a data-attribute X, the attribute in HTML would be
+     * data-{NS}-{IS}-x=""
+     * @type {string}
+     */
+    static IS = "adaptiveFormReview";
+    static bemBlock = 'cmp-adaptiveform-review';
+    static templateAttribute = 'data-cmp-review';
+    static DATA_ATTRIBUTE_VISIBLE = 'data-cmp-visible';
+    static hideFieldFromReview = ['button', 'plain-text', 'captcha', 'image'];
+    static selectors = {
+      self: "[data-" + this.NS + '-is="' + this.IS + '"]',
+      container: `.${Review.bemBlock}__container`,
+      templates: "template[data-cmp-review-fieldType]"
+    };
+    static intersectionOptions = {
+      root: null,
+      rootMargin: "0px",
+      threshold: Review.buildThresholdList(),
+    };
+    static buildThresholdList() {
+      let thresholds = [];
+      let numSteps = 20;
+      for (let i = 1.0; i <= numSteps; i++) {
+        thresholds.push(i / numSteps);
+      }
+      thresholds.push(0);
+      return thresholds;
+    }
+
+    static sanitizedValue(value, options) {
+      if(options){
```

Comment:
Recently, @im-shiv did some fixes around rich text handling via DOMPurify, could you check if that needs to be incorporated here ? If yes, let's make that code re-usable and re-use the same across multiple places.

Reply:
As discussed with @im-shiv, we don’t require any changes here.